clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1845,6 +1845,27 @@ void Sema::finalizeSYCLDelayedAnalysis(const FunctionDecl *Caller,
   }
 }
 
+bool Sema::checkAllowedSYCLInitializer(VarDecl *VD, bool CheckValueDependent) {
+  assert(getLangOpts().SYCLIsDevice &&
+         "Should only be called during SYCL compilation");
+
+  if (VD->isInvalidDecl() || !VD->hasInit() || !VD->hasGlobalStorage())
+    return true;
+
+  const Expr *Init = VD->getInit();
+  bool ValueDependent = CheckValueDependent && Init->isValueDependent();
+  bool isZeroInit = Init && !ValueDependent &&
+                    Init->isIntegerConstantExpr(Context) &&
+                    Init->EvaluateKnownConstInt(Context) == 0;
+  bool isConstantInit = Init && !ValueDependent &&
+                        Init->isConstantInitializer(Context, false);
```

Comment:
I have removed the zero initialization check; it is subsumed by the constant-initialization check.
The check for ValueDependent is needed because the helper routines called like isConstantInitializer assert when the expression is value-dependent.  When we call this routine from the instantiation context, it is no longer value dependent and this check evaluates to false.

Reply:
Got it, thanks.